### ISSUE TITLE
Pressing Return in a one-line text field doesn't submit on iOS

### ISSUE THREAD

**Issue by collinjackson** [flutter/flutter#9839]
## Steps to Reproduce

```
import 'package:flutter/material.dart';

void main() {
  runApp(new MaterialApp(
    home: new Material(
      child: new Center(
        child: new TextField(
          onSubmitted: (newValue) {
            print("submitted $newValue");
          },
        ),
      ),
    ),
  ));
}
```

I would expect that pressing Return would call print() on iOS. Right now it only does so on Android.

Originally raised on [StackOverflow](http://stackoverflow.com/questions/43806846/using-flutter-how-can-i-trigger-onsubmitted-of-textfield-on-ios), but this affects the codelab as well.

I went back in time and this bug existed on April 1, so I don't think it's a recent regression.

**Comment by cbracken** [comment]
Related #9340 (now closed but mentions the fix) and #8028 (general issue for both platforms). 

We've got the opposite problem for Android with being unable to add newlines to multi-line text fields.

**Comment by cbracken** [comment]
Rolled in #9861
